doc/api/n-api.md

```diff
@@ -5586,6 +5586,36 @@ napi_status napi_is_promise(napi_env env,
 * `[out] is_promise`: Flag indicating whether `promise` is a native promise
   object (that is, a promise object created by the underlying engine).
 
+### napi_promise_then_resolve
+<!-- YAML
+added: v17.0.0
```

Comment:
```suggestion
added: REPLACEME
```
Use the `REPLACEME` placeholder for `added`, `changes/version` or `removed` tags.